extensions/mssql/src/dashboard/serviceEndpoints.ts

```diff
@@ -64,40 +87,57 @@ export function registerServiceEndpoints(context: vscode.ExtensionContext): void
 	});
 }
 
-function getCustomEndpoint(parentEndpoint: Utils.IEndpoint, serviceName: string, serviceUrl?: string): Utils.IEndpoint {
+function getCustomEndpoint(parentEndpoint: utils.IEndpoint, serviceName: string, description: string, serviceUrl?: string): utils.IEndpoint {
 	if (parentEndpoint) {
-		let endpoint: Utils.IEndpoint = {
+		let endpoint: utils.IEndpoint = {
 			serviceName: serviceName,
-			ipAddress: parentEndpoint.ipAddress,
-			port: parentEndpoint.port,
-			isHyperlink: serviceUrl ? true : false,
-			hyperlink: 'https://' + parentEndpoint.ipAddress + ':' + parentEndpoint.port + serviceUrl
+			description: description,
+			endpoint: parentEndpoint.endpoint + serviceUrl,
+			protocol: 'https'
 		};
 		return endpoint;
 	}
 	return null;
 }
 
-function getFriendlyEndpointNames(name: string): string {
-	let friendlyName: string = name;
-	switch (name) {
+function getFriendlyEndpointNames(endpointInfo: utils.IEndpoint): string {
+	let friendlyName: string = endpointInfo.description || endpointInfo.serviceName;
+	switch (endpointInfo.serviceName) {
 		case 'app-proxy':
-			friendlyName = localize("appproxy", "Application Proxy");
+			friendlyName = localize('approxy.description', "Application Proxy");
 			break;
 		case 'controller':
-			friendlyName = localize("controller", "Cluster Management Service");
+			friendlyName = localize('controller.description', "Cluster Management Service");
 			break;
 		case 'gateway':
-			friendlyName = localize("gateway", "HDFS and Spark");
+			friendlyName = localize('gateway.description', "HDFS and Spark");
 			break;
-		case 'management-proxy':
-			friendlyName = localize("managementproxy", "Management Proxy");
+		case mgmtProxyName:
+			friendlyName = localize('mgmtproxy.description', "Management Proxy");
 			break;
-		case 'mgmtproxy':
-			friendlyName = localize("mgmtproxy", "Management Proxy");
+		case logsuiEndpointName:
+			friendlyName = kibanaDescription;
+			break;
+		case grafanaEndpointName:
+			friendlyName = grafanaDescription;
+			break;
+		case sparkHistoryEndpointName:
+			friendlyName = sparkHistoryDescription;
+			break;
+		case yarnUiEndpointName:
+			friendlyName = yarnHistoryDescription;
```

Comment:
>yarnHistoryDescription [](start = 18, length = 22)

nit: but I like to consolidate stuff like this and so you could just have getCustomEndpoint call getFriendlyEndpointNames